sbpy/spectroscopy/core.py

```diff
@@ -9,29 +9,270 @@
+
+    df : int
+        Degrees of freedom
+
+    Returns
+    -------
+    intl : `~astropy.units.Quantity`
+        Integrated line intensity at designated temperature
+
+    """
+
+    if ((energy_J-elo_J).value/(k*temp).value) and ((energy_J-elo_J).value /
+                                                    (k*300 * u.K).value) < 1:
+
+        if df == 0 or 2:
+
+            intl = lgint*(300*u.K / temp)**(2)*np.exp(-(1/temp - 1/(300*u.K))
+                                                      * elo_J/k)
+
+        else:
+
+            intl = lgint*(300*u.K/temp)**(5/2)*np.exp(-(1/temp - 1/(300*u.K))
+                                                      * elo_J/k)
+
+    else:
+
+        intl = lgint*(part300/partition)*(np.exp(-elo_J/(k*temp)) -
+                                          np.exp(-energy_J/k*temp)) / \
+               (np.exp(-elo_J/(k*300 * u.K)) - np.exp(-energy_J/(k*300*u.K)))
+
+    return intl
+
+
+def einstein_coeff(temp_estimate, transition_freq, mol_tag, vgas):
+
     """
     Einstein coefficient from molecular data
 
     Parameters
     ----------
-    frequency : `~astropy.units.Quantity`
-        Transition frequency
+    transition_freq : `~astropy.units.Quantity`
+        Transition frequency in GHz
+
+    temp_estimate : `~astropy.units.Quantity`
+        Estimated temperature in Kelvins
+
+    mol_tag : int or str
+        Molecule identifier. Make sure it is an exclusive identifier.
+
+    vgas : `~astropy.units.Quantity`
+        Gas velocity approximation in km / s
 
     Returns
     -------
-    einstein_coeff : float
+    einstein_coeff : `~astropy.units.Quantity`
         Spontaneous emission coefficient
 
-    not implemented
     """
 
+    const = constants(temp_estimate, transition_freq, mol_tag, vgas)
+
+    t_freq = const[0]
+    temp = const[1]
+    lgint = const[2]
+    part300 = const[3]
+    partition = const[4]
+    gu = const[5]
+    energy_J = const[6]
+    elo_J = const[7]
+    df = const[8]
+    h = const[9]
+    k = const[11]
+
+    intl = intensity_conversion(energy_J, elo_J, temp, k, lgint, part300,
+                                partition, df)
+
+    if (h*t_freq/(k*temp)).decompose().value and \
+            (h*t_freq/(k*300*u.K)).decompose().value < 1:
+
+        au = (lgint*t_freq
+              * (part300/gu)*np.exp(energy_J / (k*300*u.K))*(1.748e-9)).value
+
+    else:
+
+        au = (intl*(t_freq)**2 *
+              (partition/gu)*(np.exp(-elo_J/(k*temp)) -
+                              np.exp(-energy_J/(k*temp)))**(-1)
+              * (2.7964e-16)).value
+
+    return au / u.s
+
+
+def photod_rate(time, time_scale, target, id_type, observatory, format,
+                mol_tag, model):
+
+    epoch = Time(time, scale=time_scale, format=format)
+    obj = Horizons(id=target, epochs=epoch.jd, location=observatory,
+                   id_type=id_type)
+
+    try:
+        orb = obj.ephemerides()
+
+    except ValueError as error:
+
```

Comment:
`error` is not used later so no need to assign it.

Reply:
I had it assigned as a variable for debugging. I'm keeping the raise value error because it actually raises and prints good error comprehension for ephemerides and how there's more than one apparition. If I find a way to successfully apply the closest_apparition flag maybe we won't even need to raise the error.

Reply:
But I don't have it assigned as a variable anymore, cause you're right I didn't use it after I deleted the debug print statements.